packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2104,18 +2149,26 @@ class _InactiveElements {
     assert(!_locked);
     assert(!_elements.contains(element));
     assert(element._parent == null);
-    if (element._lifecycleState == _ElementLifecycle.active) {
-      _deactivateRecursively(element);
+
+    switch (element._lifecycleState) {
+      case _ElementLifecycle.active:
+        _deactivateRecursively(element);
+        // This element is only added to _elements if the whole subtree is
+        // successfully deactivated.
+        _elements.add(element);
```

Comment:
If a descendant fails to deactivate the root won't be marked as `.failed` here (it's still `.deactivated`). The caller (of `Element.deactivateChild`) may catch the exception and marked the entire subtree as failed like `inflateWidget` does.

Reply:
If the caller doesn't do that it should still be fine. The element will stuck in the inactive state forever.

Reply:
looking at the updateChild, it only mark the newChild to failed, instead of existing child (which got deactivated). or did I miss something?

Reply:
nvm, this is answered in other thread